mne/chpi.py

```diff
@@ -411,9 +416,10 @@ def _calculate_chpi_positions(raw, t_step_min=0.1, t_step_max=10.,
         time_sl = slice(max(time_sl, 0),
                         min(time_sl + hpi['n_window'], len(raw.times)))
         with use_log_level(False):
-            meg_chpi_data = raw[hpi['picks'], time_sl][0]
-        this_data = meg_chpi_data[:-1]
-        chpi_data = meg_chpi_data[-1]
+            meg_chpi_data = raw[hpi['picks'], time_sl][0] # loads good channels with hpi_stim
+        mchpi_data = np.mean(meg_chpi_data[:-1],axis=1,keepdims=True)     
```

Comment:
Hi,
I am not fully sure about a possible improvement due to the subtraction of the mean outside of the model computation. I added this because the offset is sometimes huge and I think it is numerically better if model has coefficients which are not orders of magnitude away from each other. 
Style is a difficult issue. I am a total beginner with python. Feel free to make everything more stylish ;)
I did several tests - 25 subjects, 4 blocks each. Compared that to maxfilter outputs - it is very much comparable. Shall i upload some headpos files to somewhere?
cheers, Burkhard